### PR TITLE
Improve mat64 package documentation.

### PR DESCRIPTION
**Pull request by btracey**
Explicitly mention the T() method, interface vs. concrete values, BLAS and LAPACK calls.

### CODE REVIEW

**Comment by btracey** [comment]
PTAL @kortschak 

First cut at extended documentation for mat64. I'll fix the line alignment when we're done editing.

File: mat64/doc.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// assign to the receiver, the receiver may either have the correct dimensions
+// for the result of the method or may be a zero-sized matrix. In the latter case,
+// the receiver will be modified to the correct size, allocating data if necessary.
+// The T() method is used for transposition; c.Mul(a.T(), b) computes
+// c = a^T * b. Again, if c is a zero-sized matrix it will be modified to be the
+// correct size, otherwise c must be size ac×bc, that is the number of columns in
+// a (the number of rows in a^T) by the number of columns in b.
+//
+// Many operations are performed by calling into the gonum/blas/blas64 and
+// gonum/lapack/lapack64 packages with implement BLAS and LAPACK routines
```
Comment (kortschak) [inline mat64/doc.go]: s/with/which/

**Comment by kortschak** [comment]
LGTM after lines and sub.

**Comment by kortschak** [comment]
BTW the failure is a known bug that was fixed just after beta3 was tagded.

**Pull request merged**